lib/protein/adapters/amqp_adapter/server.ex

```diff
@@ -77,14 +92,15 @@ defmodule Protein.AMQPAdapter.Server do
     end
   end
 
-  defp consume(chan, opts, meta, payload) do
+  defp consume(chan, opts, meta, payload, parent_pid) do
     server_mod = Keyword.fetch!(opts, :server_mod)
 
     {response, error} = try_process(payload, server_mod)
 
     if should_respond(response, meta), do: respond(response, chan, meta)
 
     Basic.ack(chan, meta.delivery_tag)
+    send(parent_pid, :process_finished)
```

Comment:
There's an another way for parent process to know about spawned process termination - `Process.monitor/1`. It would be the go-to approach in Elixir universe and less buggy one since termination notice is delivered always, not just after process gets to such line. It would also not require to modify the `consume` function and make it aware of parent's need to monitor it - so less coupling. Did you consider using that instead?

Reply:
Sounds good. I've implemented that